File: src/main/scala/learning_flink/course07_time_windows/HowToSetTime.scala
package learning_flink.course07_time_windows

import org.apache.flink.streaming.api.TimeCharacteristic
import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment

object HowToSetTime {
  def main(args: Array[String]): Unit = {
    val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
    // 设置处理时间类型
    env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)
  }

}

File: src/main/java/learning_flink_java/course07_time_windows/HowToSetTime.java
package learning_flink_java.course07_time_windows;

import org.apache.flink.streaming.api.TimeCharacteristic;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

public class HowToSetTime {
    public static void main(String[] args) {
        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
        env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);
//        env.setStreamTimeCharacteristic(TimeCharacteristic.ProcessingTime);
//        env.setStreamTimeCharacteristic(TimeCharacteristic.IngestionTime);
    }
}
